File: leetcode/348 Design Tic-Tac-Toe/solution.java
class TicTacToe {
	int n;
    int[][] rows; // 分别记录两个玩家n行n列以及两条对角线的情况！！！
    int[][] columns;
    int[][] diagonals;

    public TicTacToe(int n) {
        this.n = n;
		this.rows = new int[n][2];
        this.columns = new int[n][2];
        this.diagonals = new int[2][2];
    }

    public int move(int row, int col, int player) { // 仅关心当前落子所在的行，列，对角线！！！
        rows[row][player-1]++; // 行！！！
        if (rows[row][player-1] == n) return player;
       
   	    columns[col][player-1]++; // 列！！！
        if (columns[col][player-1] == n) return player;
        
		if (row == col) diagonals[0][player-1]++; // 正对角线！！！
        if (diagonals[0][player-1] == n) return player;
        
		if (row+col == n-1) diagonals[1][player-1]++; // 反对角线！！！
        if (diagonals[1][player-1] == n) return player;
        
		return 0;
    }
}

/**
 * Your TicTacToe object will be instantiated and called as such:
 * TicTacToe obj = new TicTacToe(n);
 * int param_1 = obj.move(row,col,player);
 */

File: leetcode/348 Design Tic-Tac-Toe/solution.go
type TicTacToe struct {
    N int
    Rows [][]int // 分别记录两个玩家n行n列以及两条对角线的情况！！！
    Colums [][]int
    Diagonals [][]int
}

func Constructor(n int) TicTacToe {
    rows := make([][]int, n)
    colums:= make([][]int, n)
    diagonals:= make([][]int, n)
    
    for i:=0; i<n; i++ {
        rows[i] = make([]int, 2)
        colums[i] = make([]int, 2)
        diagonals[i] = make([]int, 2)
    }

    return TicTacToe {
        N: n,
        Rows: rows,
        Colums: colums,
        Diagonals: diagonals,
    }
}


func (this *TicTacToe) Move(row int, col int, player int) int { // 仅关心当前落子所在的行，列，对角线！！！
    this.Rows[row][player-1]++
    if this.Rows[row][player-1] == this.N { // 行！！！
        return player
    }

    this.Colums[col][player-1]++
    if this.Colums[col][player-1] == this.N { // 列！！！
        return player
    }

    if row == col {
        this.Diagonals[0][player-1]++
        if this.Diagonals[0][player-1] == this.N { // 正对角线！！！
            return player
        }
    }

    if row+col == this.N-1 {
        this.Diagonals[1][player-1]++
        if this.Diagonals[1][player-1] == this.N { // 反对角线！！！
            return player
        }
    }

    return 0
}


/**
 * Your TicTacToe object will be instantiated and called as such:
 * obj := Constructor(n);
 * param_1 := obj.Move(row,col,player);
 */